alerts_backend/python/app.py

```diff
@@ -45,45 +66,145 @@
+    Returns a JSON payload of all the POSTed alerts.
+    """
+    data: Dict[str, Any] = {"posted_alerts": []}
+    with engine.connect() as conn:
+        stmt = select(aeroapi_alerts)
+        result = conn.execute(stmt)
+        conn.commit()
+        for row in result:
+            data["posted_alerts"].append(dict(row))
+
+    return jsonify(data)
+
+
+@app.route("/alert_configs")
+def get_alert_configs():
+    """
+    Function to return all the alerts that are currently configured
+    via the SQL table. Returns a JSON payload of all the configured alerts.
+    """
+    data: Dict[str, Any] = {"alert_configurations": []}
+    with engine.connect() as conn:
+        stmt = select(aeroapi_alert_configurations)
+        result = conn.execute(stmt)
+        conn.commit()
+        for row in result:
+            data["alert_configurations"].append(dict(row))
+
+    return jsonify(data)
+
+
+@app.route("/post", methods=["POST"])
+def handle_alert() -> Tuple[Response, int]:
+    """
+    Function to receive AeroAPI POST requests. Filters the request
+    and puts the necessary data into the SQL database.
+    Returns a JSON Response and also the status code in a tuple.
+    """
+    # Form response
+    r_title: str
+    r_detail: str
+    r_status: int
+    data: Dict[str, Any] = request.json
+    # Process data by getting things needed
+    processed_data: Dict[str, Any]
+    try:
+        processed_data = {
+            "long_description": data["long_description"],
+            "short_description": data["short_description"],
+            "summary": data["summary"],
+            "event_code": data["event_code"],
+            "alert_id": data["alert_id"],
+            "fa_flight_id": data["flight"]["fa_flight_id"],
+            "ident": data["flight"]["ident"],
+            "registration": data["flight"]["registration"],
+            "aircraft_type": data["flight"]["aircraft_type"],
+            "origin": data["flight"]["origin"],
+            "destination": data["flight"]["destination"],
+        }
+
+        # Check if data was inserted into database properly
+        if insert_into_table(processed_data, aeroapi_alerts) == -1:
+            r_title = "Error inserting into SQL Database"
+            r_detail = "Inserting into the database had an error"
+            r_status = 500
+        else:
+            r_title = "Successful request"
+            r_detail = "Request processed and stored successfully"
+            r_status = 200
+    except KeyError as e:
+        # If value doesn't exist, do not insert into table and produce error
+        logger.error(f"Alert POST request did not have one or more keys with data. Will process but will return 400: {e}")
+        r_title = "Missing info in request"
+        r_detail = "At least one value to insert in the database is missing in the post request"
+        r_status = 400
+
+    return jsonify({"title": r_title, "detail": r_detail, "status": r_status}), r_status
```

Comment:
Appreciate the work done here for replying with a coherent error message if something goes wrong.